src/Components/Forms/test/FieldIdentifierTest.cs

```diff
@@ -89,6 +102,19 @@ public void SameContentsProduceSameHashCodesAndEquality()
             Assert.True(fieldIdentifier1.Equals(fieldIdentifier2));
         }
 
+        [Fact]
+        public void SameContents_WithOverridenEqualsAndGetHashCode_ProduceSameHashCodesAndEquality()
+        {
+            // Arrange
+            var model = new EquatableModel();
+            var fieldIdentifier1 = new FieldIdentifier(model, nameof(EquatableModel.Property));
+            var fieldIdentifier2 = new FieldIdentifier(model, nameof(EquatableModel.Property));
```

Comment:
```suggestion
            model.Property = "changed value"; // To show it makes no difference if the overridden `GetHashCode` result changes
            var fieldIdentifier2 = new FieldIdentifier(model, nameof(EquatableModel.Property));
```

Not sure this suggestion really adds that much extra value so please feel free to ignore it if you prefer.